hummingbot/connector/exchange/mexc/mexc_api_user_stream_data_source.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+import asyncio
+import hashlib
+import json
+from urllib.parse import urlencode
+
+import aiohttp
+import aiohttp.client_ws
+
+import logging
+
+from typing import (
+    Optional,
+    AsyncIterable,
+    List,
+    Dict,
+    Any
+)
+
+from hummingbot.connector.exchange.mexc.constants import MEXC_WS_URL_PUBLIC
+from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
+from hummingbot.logger import HummingbotLogger
+from hummingbot.connector.exchange.mexc.mexc_auth import MexcAuth
+
+import time
+
+from websockets.exceptions import ConnectionClosed
+
+
+class MexcAPIUserStreamDataSource(UserStreamTrackerDataSource):
+    _mexcausds_logger: Optional[HummingbotLogger] = None
+    MESSAGE_TIMEOUT = 300.0
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._mexcausds_logger is None:
+            cls._mexcausds_logger = logging.getLogger(__name__)
+
+        return cls._mexcausds_logger
+
+    def __init__(self, mexc_auth: MexcAuth, trading_pairs: Optional[List[str]] = []):
+        self._current_listen_key = None
+        self._current_endpoint = None
+        self._listen_for_user_stram_task = None
+        self._last_recv_time: float = 0
+        self._auth: MexcAuth = mexc_auth
+        self._trading_pairs = trading_pairs
+        super().__init__()
```

Comment:
As much as possible, lets use the same `_shared_client: aiohttp.ClientSession` from `MexcExchange`.